### ISSUE TITLE
no-unused-vars fails to mark interfaces as used when destructuring in eg. a function parameter

### ISSUE THREAD

**Issue by lostfictions** [nzakas/eslint-plugin-typescript#82]
With these versions:
```
"dependencies": {
  "eslint": "^4.8.0",
  "eslint-plugin-typescript": "^0.8.0",
  "typescript-eslint-parser": "^8.0.0",
  "typescript": "^2.5.3"
}
```

and these rules:
```
"rules": {
    "no-unused-vars": "warn",
    "typescript/no-unused-vars": "warn"
}
```

and this test case:

`test.ts`
```typescript
import { ReproInterface } from './test2'

const x = ({ a } : ReproInterface) => a
```

`test2.ts`
```typescript
export interface ReproInterface {
  a : number
}
```

In `test.ts`, `ReproInterface` is incorrectly marked as unused. If the function parameter is not destructured, the warning goes away.

**Comment by weirdpattern** [comment]
Hi @lostfictions, thank you for reporting this. I will take a look at this some time tomorrow...
